src/orange/orangeinp/detail/VolumeBuilder.hh

```diff
@@ -38,24 +40,83 @@ class VolumeBuilder
     //!@}
 
   public:
-    // Construct with unit builder and volume name
+    // Construct with unit builder (and volume name??)
     explicit VolumeBuilder(CsgUnitBuilder* ub);
 
+    //// ACCESSORS ////
+
     //!@{
     //! Access the unit builder for construction
     CsgUnitBuilder const& unit_builder() const { return *ub_; }
     CsgUnitBuilder& unit_builder() { return *ub_; }
     //!@}
 
-    //! Access the local-to-global transform during construction
-    VariantTransform const& local_transform() const { return local_trans_; }
+    // Access the local-to-global transform during construction
+    VariantTransform const& local_transform() const;
+
+    //// MUTATORS ////
 
     // Add a region to the CSG tree
     NodeId insert_region(Metadata&& md, Joined&& j, BoundingZone&& bzone);
 
+    // Apply a transform within this scope
+    [[nodiscard]] PopVBTransformOnDestruct
+    make_scoped_transform(VariantTransform const& t);
+
   private:
+    //// DATA ////
+
     CsgUnitBuilder* ub_;
-    VariantTransform local_trans_;  //!< DUMMY for now
+    std::vector<TransformId> transforms_;
+
+    //// PRIVATE METHODS ////
+
+    // Add a new variant transform
+    void push_transform(VariantTransform&& vt);
+
+    // Pop the last transform, used only by PopVBTransformOnDestruct
+    void pop_transform();
+
+    //// FRIENDS ////
+
+    friend class PopVBTransformOnDestruct;
+};
+
+//---------------------------------------------------------------------------//
+//! Implementation-only RAII helper class for VolumeBuilder
```

Comment:
Could you add some additional info about the objective of this class w.r.t. scoping?